source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,313 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide you will learn how to perform **compound operations** with the
+MongoDB Java driver. 
+
+Compound operations consist of a read and write operation performed as one
+**atomic operation**. An atomic operation is an operation which either completes
+entirely, or does not complete at all. Atomic operations cannot partially complete.
```

Comment:
Based on your current definition, it's implied that atomic operations cannot partially complete.

```suggestion
entirely, or does not complete at all.
```

Reply:
Thanks for the note. As atomic operations and concurrency are a notoriously tricky concept, I would like to leave this in. I think it will benefit readers to have the corollary explicitly stated as the definition is dense, nuanced, and jargon heavy.